### ISSUE TITLE
Significant blank margins added around plotted figure when calling geom_cladlabel

### ISSUE THREAD

**Issue by megaraptor1** [YuLab-SMU/ggtree#388]
I'm having an issue with the geom_cladlab function in ggtree. I am trying to plot my tree into an RMarkdown html document, but I'm noticing that adding a geom_cladlab call results in the tree being shrunk significantly and a large amount of blank white space being added around the figure. At first I thought this was a consequence of calling geom_cladlab to add objects that would obviously extend the maximum dimensions of the shape, but when I examined the object itself I found that there was a significant amount of white space that was just dead space and could not be explained by the addition of clade labels adjusting the maximum dimensions of the object.

This is a bit of an issue for me as my tree is quite large and shrinking it down in this manner results in the taxon labels being difficult to read. It also makes it difficult to save the tree at the appropriate size and resolution for publication, even if I try to trim the white space off in Photoshop.

Here is some code that replicates my problem. I am including RMarkdown formatting using the rmdformats package because it makes the printed figures clickable and clicking on the figure really shows how much of the added white space on the margins has nothing in it.

---
output:
  rmdformats::html_clean:
    highlight: haddock
    toc: FALSE
    thumbnails: FALSE
    code_folding: show
    df_print: paged
---

```{r setup, include=FALSE}
library(knitr)
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse);library(treeio);library(rmdformats);library(ape);library(phytools);library(ggtree);library(curl)
```

```{r}
rodent.tree<-read.nexus(curl("https://raw.githubusercontent.com/megaraptor1/mydata/main/engelmanData.tree"))
d<-read.csv(curl("https://raw.githubusercontent.com/megaraptor1/mydata/main/traitdata.csv"))
df<- full_join(rodent.tree, d, by = 'node')
```

Tree without clade labels:

```{r,fig.width=24,fig.height=24}
ggtree(df,layout="circular",size=2.5)+
  geom_tree(aes(color=trait),continuous=T,size=2)+
  geom_tiplab(color="black",size=4,fontface='bold.italic')+
  scale_color_gradientn(colours=c("red", 'orange', 'green', 'cyan', 'blue'))+
  theme(legend.position=c(.9,.93),legend.key.size = unit(1, "cm"),legend.title=element_text(size=24),legend.text=element_text(size=24))
```

Tree with clade labels:

```{r,fig.width=24,fig.height=24}
ggtree(df,layout="circular",size=2.5)+
  geom_tree(aes(color=trait),continuous=T,size=2)+
  geom_tiplab(color="black",size=4,fontface='bold.italic')+
geom_cladelabel(offset=24,node=820,label="Anomaluromorpha",color="red",hjust=1,align=T)+
geom_cladelabel(offset=24,node=826,label="Myomorpha",color="blue",align=T)+
  geom_cladelabel(offset=24,node=685,label="Hystricomorpha",hjust=1,color="green",align=T)+
geom_cladelabel(offset=24,node=1064,label="Castorimorpha",color="orange",hjust=1,align=T)+
geom_cladelabel(offset=24,node=553,label="Sciuromorpha",color="cyan",align=T)+
  scale_color_gradientn(colours=c("red", 'orange', 'green', 'cyan', 'blue'))+
  theme(legend.position=c(.9,.93),legend.key.size = unit(1, "cm"),legend.title=element_text(size=24),legend.text=element_text(size=24))
```

To produce a more visual example, here is what I get when I plot the graph with all clade labels angled perpendicular to the tree tips and thus at minimum width (with a black border plotted around the graph using ggplot2 to show its margins)

![example](https://user-images.githubusercontent.com/74024763/113601159-fb9cd500-960e-11eb-933f-6ec87e89b241.jpeg)

And what happens when I plot the same graph setting "label = NA" to remove all labels in geom_cladlab.

![example1](https://user-images.githubusercontent.com/74024763/113601169-fe97c580-960e-11eb-9b08-0765a691fad2.jpeg)

Compared to if no geom_cladelab argument is called:

![example2](https://user-images.githubusercontent.com/74024763/113601480-64844d00-960f-11eb-8e56-1d9336b54d41.jpeg)

I looked at the vignettes and handbook for ggtree as well as on Google and Stack Overflow, and I didn't see anything about how to fix this.

On a related note, do you know if it is possible to set the offset option to take into account the call for geom_tiplab? I think part of the problem is trying to manually estimate the value of the offset call such that it does not overlap with the geom_tiplab call, and I think that might be increasing the margins of the plot. I was trying to create a reprex example using the anole SVL tree, but I was having trouble because I could not make a geom_cladelab call that would produce a reproducible example. I had to set the offset by trial and error, and I noticed that if I made a call greater than 0.16 it cut off the clade label name, and above 0.17 it also cut off the clade label.

Overall, I think having some kind of setting for offset or geom_cladelabel to automatically scale the offset for geom_cladelabel to the outer edge of the taxon names would be really useful given how many phylogenetic trees combine taxonomic names at the tips of the tree with labels for broader groups of interest. It would really help to cut down on trial and error scaling, especially if the tree is going to be printed at different sizes. I've noticed that trying to adjust print sizes both in this and in phytools' plot options often results in needing to rescale the tree using trial and error all over again.

See attached code for examples of what I mean with the disappearing clade label issue. The demarked clade of interest is the MRCA of "equestris" and "baracoae", but it's a bit hard to see because I cannot plot it beyond an offset of 0.18 and thus I cannot plot it beyond the tip labels:

```{r}
anole.tree<-read.tree("http://www.phytools.org/eqg2015/data/anole.tre")
svl <- read.csv("http://www.phytools.org/eqg2015/data/svl.csv",
                row.names=1)
svl <- as.matrix(svl)[,1]
fit <- fastAnc(anole.tree,svl,vars=TRUE,CI=TRUE)
td <- data.frame(node = nodeid(anole.tree, names(svl)),
                 trait = svl)
nd <- data.frame(node = names(fit$ace), trait = fit$ace)
d <- rbind(td, nd)
d$node <- as.numeric(d$node)
tree <- full_join(anole.tree, d, by = 'node')
```

```{r fig.cap="Tree where offset = 0.16, clade label appears with shape and clade name"}
ggtree(tree, aes(color=trait), layout = 'circular', 
       ladderize = FALSE, continuous = TRUE, size=2) +
  scale_color_gradientn(colours=c("red", 'orange', 'green', 'cyan', 'blue')) +
  geom_cladelabel(offset=.16,node=getMRCA(anole.tree,tip=c("equestris","noblei")),label="Clade",color="red",align=T)+
  geom_tiplab(hjust = -.1) + 
  xlim(0, 1.2) + 
  theme(legend.position = c(.05, .85)) 
```

![anole1](https://user-images.githubusercontent.com/74024763/113601684-a44b3480-960f-11eb-83f5-bf4a7ac7ea36.jpeg)

```{r fig.cap="Tree where offset = 0.17, clade label appears but clade name does not}
ggtree(tree, aes(color=trait), layout = 'circular', 
             ladderize = FALSE, continuous = TRUE, size=2) +
  scale_color_gradientn(colours=c("red", 'orange', 'green', 'cyan', 'blue')) +
  geom_cladelabel(offset=.17,node=getMRCA(anole.tree,tip=c("equestris","noblei")),label="Clade",color="red",align=T)+
  geom_tiplab(hjust = -.1) + 
  xlim(0, 1.2) + 
  theme(legend.position = c(.05, .85)) 
```

![anole2](https://user-images.githubusercontent.com/74024763/113601699-aca36f80-960f-11eb-96d3-3c3a69bf3105.jpeg)

```{r fig.cap="Tree where offset = 0.18, both clade label and clade name do not appear}
ggtree(tree, aes(color=trait), layout = 'circular', 
             ladderize = FALSE, continuous = TRUE, size=2) +
  scale_color_gradientn(colours=c("red", 'orange', 'green', 'cyan', 'blue')) +
  geom_cladelabel(offset=.18,node=getMRCA(anole.tree,tip=c("equestris","noblei")),label="Clade",color="red",align=T)+
  geom_tiplab(hjust = -.1) + 
  xlim(0, 1.2) + 
  theme(legend.position = c(.05, .85)) 
```

![anole3](https://user-images.githubusercontent.com/74024763/113601714-b036f680-960f-11eb-8602-af785a36fade.jpeg)

I even noticed this issue was present to some degree when I just plotted a random tree, in this case there is extra white space around the graph, though not as much as in the first example, the circular bar for geom_cladlabel is not showing up, and now there is an additional issue in that the vjust call is not being recognized.

```{r}
set.seed(200)
tree<-force.ultrametric(rtree(550))
ggtree(tree,layout="circular")+
  geom_cladelabel(offset=0.1,node=1,horizontal=FALSE,angle="auto",label="Clade2",color="blue",align=T)+
  geom_cladelabel(offset=0.1,node=300,horizontal=FALSE,angle="auto",label="Clade3",color="cyan",align=T)+
  geom_cladelabel(offset=0.1,node=150,horizontal=FALSE,angle="auto",label="Clade4",color="green",align=T)+
  geom_cladelabel(offset=0.1,node=100,horizontal=FALSE,angle="auto",label="Clade5",color="orange",align=T)+
  geom_cladelabel(offset=0.1,node=400,horizontal=FALSE,angle="auto",label="Clade6",color="purple",align=T)+
  geom_cladelabel(offset=0.1,node=200,horizontal=FALSE,angle="auto",label="Clade",color="red",align=T)+
  theme(panel.background = element_rect(colour = "black", size=0.5))```
```

![rtree](https://user-images.githubusercontent.com/74024763/113602611-d27d4400-9610-11eb-9090-8ed1883e4bda.jpeg)

Indeed, this tree is some ways is even weirder because it does not produce the arc denoting the clade in addition to plotting the extra margins. This is really surprising given that this is just a randomly generated tree and it does this. In fact if I adjust offset to plot the clade labels inside the tree it still produces the extra margins...

![inside](https://user-images.githubusercontent.com/74024763/113602809-17a17600-9611-11eb-8c97-9ecb3bd13813.jpeg)

### Prerequisites

+ [x] Have you read [Feedback](https://guangchuangyu.github.io/ggtree/#feedback) and follow the [guide](https://guangchuangyu.github.io/2016/07/how-to-bug-author/)?
	* [x] make sure your are using the latest release version
	* [x] read the [documents](https://yulab-smu.github.io/treedata-book/)
	* [x] google your question/issue

### Describe you issue

* [x] Make a reproducible example (*e.g.* [1](https://gist.github.com/talonsensei/e1fad082657054207f249ec98f0920eb))
* [x] your code should contain comments to describe the problem (*e.g.* what expected and actually happened?)

### Ask in right place

* [x] for bugs or feature requests, post here (github issue)
* [ ] for questions, please post to [google group](https://groups.google.com/forum/#!forum/bioc-ggtree)

**Comment by megaraptor1** [comment]
I did some more experimentation with the code and it turns out even setting label=NA for the geom_cladlab calls does not remove the large white space surrounding the figure. It still results in a significant (~5 cm) white blank area surrounding the figure itself.

**Comment by xiangpin** [comment]
I think this is not the problem of `geom_cladelabel`. You can try adding the following code since you are using markdown.

```
 library(knitr)
 knit_hooks$set(crop = hook_pdfcrop)
 opts_chunk$set(crop = TRUE)
```

**Comment by megaraptor1** [comment]
I'm pretty sure this isn't the result of `geom_cladelabel` or `geom_cladelab`. The pictures I attached were ones I saved in the base R program using `ggsave` without using markdown. It seems like the `geom_cladelab` function by itself creates the addition of an additional white margin around the figure, even if the rest of the figure is set to `NA` or spaced so it is inside the tree rather than outside.
